src/ucs/datastruct/mpool.c

```diff
@@ -214,6 +213,8 @@ void ucs_mpool_grow(ucs_mpool_t *mp, unsigned num_elems)
     chunk->num_elems = ucs_min(data->quota, (chunk_size - chunk_padding - sizeof(*chunk)) /
                        ucs_mpool_elem_total_size(data));
 
+    chunk->num_elems = ucs_min(chunk->num_elems, num_elems);
```

Comment:
why needed? it causes memory waste in UCT memory pools
maybe instead chuck init could calc/get the actual number of elements?